evennia/scripts/tests.py

```diff
@@ -88,3 +159,132 @@ def test_start__delay(self, MockLoopingCall):
         loopcall.__call__.assert_not_called()
         self.assertEqual(loopcall.interval, 20)
         loopcall._scheduleFrom.assert_called_with(121)
+
+    def test_start_invalid_interval(self):
+        """ Test the .start method with interval less than zero """
+        with self.assertRaises(ValueError):
+            callback = mock.MagicMock()
+            loopcall = ExtendedLoopingCall(callback)
+            loopcall.start(-1, now=True, start_delay=None, count_start=1)
+
+    def test__call__when_delay(self):
+        """ Test __call__ modifies start_delay and starttime if start_delay was previously set """
+        callback = mock.MagicMock()
+        loopcall = ExtendedLoopingCall(callback)
```

Comment:
Since the main task here is not mocked, I think this looping call should be explicitly killed at the end of each test; otherwise there is a list of the async component firing outside of the scope of the test, which is never a good idea.

Reply:
In this case, the looping call is never put on repeat using .start(...) so I'm guessing it should be fine in this case. It can't be terminated with .stop() as it isn't running.